libdevice/include/assert.h

```diff
@@ -0,0 +1,76 @@
+//==--------- devicelib-assert.h - wrapper definition for C assert ---------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __DEVICELIB_ASSERT_H__
+#define __DEVICELIB_ASSERT_H__
+
+#include_next "assert.h"
```

Comment:
I think this is not required if we rename this header to something like __spir_assert.h and include it any time compilation for spir target is done. This can be achieved if we move the header to the compiler headers as suggested above.

Reply:
So clang driver will have to add `-include $CLANG_LIB/__spir_assert.h` options for all device headers we support - is that what you propose? 

For the record, there are two minor disadvantages:
  - we have to patch clang driver to add a new header
  - all headers must be included - currently they contain ~500 LOC combined

Reply:
> So clang driver will have to add `-include $CLANG_LIB/__spir_assert.h` options for all device headers we support - is that what you propose?

Something like this: https://github.com/intel/llvm/blob/sycl/clang/lib/Driver/ToolChains/Cuda.cpp#L247

> 
> For the record, there are two minor disadvantages:
> 
> * we have to patch clang driver to add a new header
> * all headers must be included - currently they contain ~500 LOC combined

I see these as disadvantages.

Reply:
> 
> > For the record, there are two minor disadvantages:
> > 
> > * we have to patch clang driver to add a new header
> > * all headers must be included - currently they contain ~500 LOC combined
> 
> I see these as disadvantages.

Sorry, it is not clear whether you're in favor of `__spir_assert.h` or `assert.h + #include_next`.

Reply:
Sorry, I've made a typo in this comment. I *don't think* these the items are disadvantages of the approach.